### PR TITLE
Merge conflict fix

### PR DESCRIPTION
**Pull request by aarondr77**
# Changes

- [x] I ran all the tests to make sure I didn't break anything.

### CODE REVIEW

File: src/taskpane/components/MergeComponents/MergeConflictScreen.js [hunk 1]
```diff
@@ -57,7 +57,7 @@ export default class MergeConflictScreen extends React.Component {
     let usingDefault = false;
 
     // TODO: Check this value without converting to a string
-    const isDefaultPersonal = document.getElementById("default-toggle").getAttribute('aria-checked').toString();
+    const isDefaultPersonal = document.getElementById("default-toggle").getAttribute('aria-checked').toString() === "true";
 
     this.setState({default: isDefaultPersonal === "true" ? "your changes" : "your collaborator's changes"});
```
Comment (naterush) [inline src/taskpane/components/MergeComponents/MergeConflictScreen.js]: No more string comparison.

File: src/tests/suites/mergeConflictsTests.js [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+
+    // Check that there is no merge conflict
+    assert.deepEqual(mergeResult, expected, "there was a merge conflict");
+
+    // Then resolve the merge conflicts
+    const resolutions = {"Sheet2": [{cellOrRow: "A1", value: "P3"}, {cellOrRow: "B1", value: "P4"} ], "Sheet1": [{cellOrRow: "A1", value: "P1"}, {cellOrRow: "B1", value: "P2"}]};
+    await runResolveMergeConflicts(resolutions);
+
+    // Check that merge conflicts are resolved correctly
+    const updatedValue1A1= (await runOperation(getFormulas, "Sheet1", "A1"))[0][0];
```
Comment (naterush) [inline src/tests/suites/mergeConflictsTests.js]: Get the values in one operation and pick them off. Anything we do to not have horribly slow tests...

File: src/tests/suites/mergeFormattingTests.js [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+    
+    // Check for boldness
+    Excel.run(async function (context) {
+        var sheet = context.workbook.worksheets.getItem("Sheet1");
+    
+        var range = sheet.getRange("A1");
+        range.load("format/font/bold");
+        await context.sync();
+
+        assert.equal(range.format.font.bold, true,  "A1 should be bold");
```
Comment (naterush) [inline src/tests/suites/mergeFormattingTests.js]: This assert needs to be outside the Excel.run

File: src/tests/suites/mergeFormattingTests.js [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+    
+        return await context.sync();
+    });
+    
+    // Perform a merge
+    const g = getGlobal();
+    await g.merge();
+    
+    // Check for boldness
+    Excel.run(async function (context) {
```
Comment (naterush) [inline src/tests/suites/mergeFormattingTests.js]: You have to wait in the Excel.run to finish; it returns a promise.

File: src/tests/suites/mergeFormattingTests.js [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+    // Load scenario
+    const fileContents = scenarios["diffSimple"].fileContents;
+    await runReplaceFromBase64(fileContents);
+
+    // Give time for files to update properly 
+    await new Promise(resolve => setTimeout(resolve, 2000));
+    
+
+    // Bold A1:B3
+    Excel.run(async function (context) {
```
Comment (naterush) [inline src/tests/suites/mergeFormattingTests.js]: same here.

File: src/tests/suites/mergeFormattingTests.js [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+    
+        await context.sync();
+    });
+    
+    // Perform a merge
+    const g = getGlobal();
+    await g.merge();
+    
+    // Check for boldness
+    Excel.run(async function (context) {
```
Comment (naterush) [inline src/tests/suites/mergeFormattingTests.js]: and here

**Review by naterush — CHANGES_REQUESTED** [review]
Minor review for now. Will review in depth after changes.

**Comment by aarondr77** [comment]
ooo shoot. I think when I rebased, I forgot to redo some of those fixes. Will do now.

File: src/tests/suites/mergeConflictsTests.js [hunk 7]
```diff
@@ -0,0 +1,173 @@
...
+    const g = getGlobal();
+    const mergeResult = await g.merge();
+    console.log(mergeResult);
+    assert.equal(mergeResult.status, mergeState.MERGE_CONFLICT, "Should be a merge conflict");
+
+
+    // Check that the conflict is correct
+    const mergeConflictData = mergeResult.mergeConflictData;
+    
+    assert.equal(mergeConflictData[0].sheet, "Sheet2", "should contain conflicts on Sheet 2");
```
Comment (naterush) [inline src/tests/suites/mergeConflictsTests.js:88]: For now, this is fine, but it seems better to:
1. Make an object and do the comparison all at once
2. Have a test running that can helpfully display that error to you.

I'll open an issue now.

**Review by naterush — APPROVED** [review]
This looks good to me. LGTM.

**Pull request merged**